packages/client/modules/teamDashboard/components/AgendaList/AgendaList.tsx

```diff
@@ -50,18 +50,19 @@ const AgendaList = (props: Props) => {
 
   const onDragEnd = useEventCallback((result) => {
```

Comment:
+1 I think TypeScript can save us here. We can get the `DropResult` type from `react-beautiful-dnd` and do `result: DropResult`. That tells us that `destination` can be undefined while `source` won't be. 

The bug happened on line 53: `agendaItems[destination.index]` as `destination` could be undefined. This new solution works because `agendaItems[destination.index]` is now happening after `!destination` on line 54. 

An alternative solution could be:

```
    const {source, destination} = result
    if (!destination) return
    const destinationItem = agendaItems[destination.index]
    const sourceItem = agendaItems[source.index]
    if (
      destination.droppableId !== AGENDA_ITEM ||
      source.droppableId !== AGENDA_ITEM ||
      destination.index === source.index ||
      !destinationItem ||
      !sourceItem
    ) {
      return
    }
```

Reply:
I like this a lot!